Pal/src/host/Linux-SGX/tools/ra-tls/secret_prov_attest.c

```diff
@@ -78,19 +97,31 @@ int secret_provision_start(const char* in_servers, const char* in_ca_chain_path,
     char* connected_addr = NULL;
     char* connected_port = NULL;
 
-    mbedtls_ctr_drbg_init(&g_ctr_drbg);
-    mbedtls_entropy_init(&g_entropy);
-    mbedtls_x509_crt_init(&g_verifier_ca_chain);
+    mbedtls_ctr_drbg_context ctr_drbg;
+    mbedtls_entropy_context entropy;
+    mbedtls_x509_crt verifier_ca_chain;
+    mbedtls_pk_context my_ratls_key;
+    mbedtls_x509_crt my_ratls_cert;
+
+    mbedtls_net_context* verifier_fd = malloc(sizeof(*verifier_fd));
+    mbedtls_ssl_config*  conf        = malloc(sizeof(*conf));
+    mbedtls_ssl_context* ssl         = malloc(sizeof(*ssl));
+    if (!verifier_fd || !conf || !ssl) {
```

Comment:
if verifier_fd was allocated successfully, but ssl failed to be allocated, then return -ENOMEM here will lead to memory leak.